include/nuttx/sched_note.h

```diff
@@ -236,17 +248,24 @@ struct note_spinlock_s
 #ifdef CONFIG_SCHED_INSTRUMENTATION_SYSCALL
 /* This is the specific form of the NOTE_SYSCALL_ENTER/LEAVE notes */
 
+#define MAX_SYSCALL_ARGS  6
+#define SIZEOF_NOTE_SYSCALL_ENTER(n) (sizeof(struct note_common_s) + \
+                                      sizeof(uint8_t) + sizeof(uint8_t) + \
+                                      (sizeof(uintptr_t) * (n)))
+
 struct note_syscall_enter_s
 {
-  struct note_common_s nsc_cmn; /* Common note parameters */
-  int nsc_nr;                   /* System call number */
+  struct note_common_s nsc_cmn;                           /* Common note parameters */
+  uint8_t nsc_nr;                                         /* System call number */
+  uint8_t nsc_argc;                                       /* Number of system call arguments */
+  uint8_t nsc_args[sizeof(uintptr_t) * MAX_SYSCALL_ARGS]; /* System call arguments */
 };
 
 struct note_syscall_leave_s
 {
-  struct note_common_s nsc_cmn; /* Common note parameters */
-  uintptr_t nsc_result;         /* Result of the system call */
-  int nsc_nr;                   /* System call number */
+  struct note_common_s nsc_cmn;          /* Common note parameters */
+  uint8_t nsc_nr;                        /* System call number */
+  uint8_t nsc_result[sizeof(uintptr_t)]; /* Result of the system call */
```

Comment:
It's better to move arg/result saving change to another patch, so we can merge it to mainline directly. Also please don't forget to update the apps/system/sched_note/.